docs/core/common/assets/js/api/core/components.md

```diff
@@ -0,0 +1,116 @@
+	|------------------------------|--------------------------------|-----------------------------------------------|------------------------------|
+	| `$e.components.getAll()`     |                                | `{array.<string>}`                            | Receive all components.
+	| `$e.components.register()`   | `{ComponentBase}` *component*  | `{ComponentBase}` *component*                 | Register a component.
+	| `$e.components.get()`        | `{String}` *id*                | `{ComponentBase}` *component*                 | Get component instance by id.
+	| `$e.components.getActive()`  |                                | `{Object.<ComponentBase>}` *activeComponents* | Get active components.
+	| `$e.components.activate()`   | `{String}` *namespace*         |                                               | Activate component.
+	| `$e.components.inactivate()` | `{String}` *namespace*         |                                               | Deactivate component.
+	| `$e.components.isActive()`   | `{String}` *namespace*         | `{Boolean}` *isActive*                        | Is component active.
+     > **Note:** to see all the components please. please visit [`{$e.commands.getAll()}`](commands-methods/getall.md)**method**
+
+* **Examples**:
+    ```javascript
+     // Example create and register new component, available to run in the console does not depends on anything else.
+     class CustomComponent extends $e.modules.ComponentBase {
+        getNamespace() {
+            return 'custom-component';
+        }
+     
+        defaultCommands() {
+            // Object of all the component commands.
+            return {
+              // 'example' command.   
+              example: ( args ) => {
+                  // Output command args to console.
+                  console.log( 'ExampleCommand: ', args );
+           
+                  // Return object as example.
+                  return {
+                      example: 'result from ExampleCommand',
+                  };
+              },
+            };
+        }
+     }
+     
+     // Register the new component.
+     $e.components.register( new CustomComponent() );
+     
+     // Run's 'example' command from 'custom-component'.
+     result = $e.run( 'custom-component/example', {
+        property: 'value',
+     } );
+     
+     // Output command run result.
+     console.log( 'e-components-eg-1-result: ', result );
+    
+    ```
+## Guidelines, conventions & file's structure
+  * You can view your component as namespace, that holds your [commands](../core/commands.md#guidelines-conventions--files-structure), [hooks](../core/hooks.md#guidelines-conventions--files-structure), [routes](#UPDATE_WHEN_READY), [tabs](#UPDATE_WHEN_READY), [shortcuts](#UPDATE_WHEN_READY) & [utils](#UPDATE_WHEN_READY).
+  * Component class file should be named `component.js`
+  * Component folder name should be named as component namespace or sub component namespace.
```

Comment:
```suggestion
  * Component folder name should be named as a component namespace or a sub-component namespace.
```

Reply:
We have to fix some of our current files structure according to these conventions

Reply:
After we done with 'conventions' i will add to my self a task: fix new conventions in existing code.